### PR TITLE
Adding cluster pool claim step to aws-account-operator ci

### PR DESCRIPTION
**Pull request by rajdeepc2792**
Modifying a non operating step to existing CI-Operator config for [openshift/aws-account-operator].
Adding cluster claim to `integration-test` step, still a noop step.
(https://github.com/openshift/aws-account-operator)
Jira Issue - https://issues.redhat.com/browse/OSD-12223

### CODE REVIEW

File: ci-operator/config/openshift/aws-account-operator/openshift-aws-account-operator-master.yaml [hunk 1]
```diff
@@ -42,10 +42,32 @@ tests:
   commands: make validate
   container:
     from: src
-- as: test-int
-  commands: echo "Test Integration WIP"
-  container:
-    from: src
+- as: integration-test
```
Comment (macgregor) [inline ci-operator/config/openshift/aws-account-operator/openshift-aws-account-operator-master.yaml:45]: is there a way to make this step to not prevent merging if it fails? that would be convenient while we develop this.

Comment (rajdeepc2792) [reply ci-operator/config/openshift/aws-account-operator/openshift-aws-account-operator-master.yaml:45]: According to this link, assigning `optional` to `true` will not prevent merging if the CI step fails:- https://docs.ci.openshift.org/docs/architecture/ci-operator/#pre-submit-tests

I will add `optional: true` to this PR.

File: ci-operator/config/openshift/aws-account-operator/openshift-aws-account-operator-master.yaml [hunk 2]
```diff
@@ -42,10 +42,32 @@ tests:
...
+  cluster_claim:
+    architecture: amd64
+    cloud: aws
+    owner: openshift-ci
+    product: ocp
+    timeout: 2h0m0s
```
[the remark this answers is not in the document]
Comment (rajdeepc2792) [reply ci-operator/config/openshift/aws-account-operator/openshift-aws-account-operator-master.yaml]: changed timeout value to `15m0s`.

**Comment by macgregor** [comment]
/lgtm 

@jharrington22 @NautiluX can one of you approve, our two approvers on the team are on PTO

File: ci-operator/jobs/openshift/aws-account-operator/openshift-aws-account-operator-master-presubmits.yaml [hunk 3]
```diff
@@ -130,9 +134,15 @@ presubmits:
...
             cpu: 10m
         volumeMounts:
+        - mountPath: /secrets/ci-pull-credentials
+          name: ci-pull-credentials
+          readOnly: true
         - mountPath: /secrets/gcs
           name: gcs-credentials
           readOnly: true
+        - mountPath: /secrets/hive-hive-credentials
+          name: hive-hive-credentials
```
Comment (dkeohane) [inline ci-operator/jobs/openshift/aws-account-operator/openshift-aws-account-operator-master-presubmits.yaml:144]: Why not just hive-credentials?

Comment (rajdeepc2792) [reply ci-operator/jobs/openshift/aws-account-operator/openshift-aws-account-operator-master-presubmits.yaml:144]: This secret is part of the default secrets generated as a part of https://github.com/openshift/release/blob/889ee97c26570aa6860fdeb24acfe5509d040c88/core-services/ci-secret-bootstrap
secret name present here:- https://github.com/openshift/release/blob/889ee97c26570aa6860fdeb24acfe5509d040c88/core-services/ci-secret-bootstrap/_config.yaml#L1967

`*-presubmits.yaml` is an auto-generated yaml using [openshift-aws-account-operator-master.yaml file](https://github.com/openshift/release/pull/30474/files/4e00249352f6622d783858424f5175508ddf37c1#diff-bc5eb24b9029beee2b5ad639283be862b11323d209cbfa811e2896e0fcc1b28b)

**Pull request merged**